snap/README.md

```diff
@@ -0,0 +1,30 @@
+# EdgeX REST Device Service Snap
+[![snap store badge](https://raw.githubusercontent.com/snapcore/snap-store-badges/master/EN/%5BEN%5D-snap-store-black-uneditable.png)](https://snapcraft.io/edgex-device-rest)
+
+This folder contains snap packaging for the EdgeX REST Protocol Device Service Snap
+
+The snap currently supports both `amd64` and `arm64` platforms.
+
+
+## Snap configuration
+
+Device services implement a service dependency check on startup which ensures that all of the runtime dependencies of a particular service are met before the service transitions to active state.
+
+The logic implements a basic retry loop with a maximum duration and sleep interval. If the dependent services are not available, the service sleeps for the defined interval (default: 1s) and then tries again up to a maximum duration (default: 60s). These values can be overridden with the following commands:
```

Comment:
It might be a good idea to offer an explanation of when and/or why someone would need to use these config settings. See my [device-camera-review](https://github.com/edgexfoundry/device-camera-go/pull/66#pullrequestreview-542325627) for suggested wording.

Reply:
Added "Snapd doesn't support orchestration between services in different snaps. It is therefore possible on a reboot for a device service to come up faster than all of the required services running in the main edgexfoundry snap. If this happens, it's possible that the device service repeatedly fails startup, and if it exceeds the systemd default limits, then it might be left in a failed state. This situation might be more likely on constrained hardware (e.g. RPi)."